doc/contract-update.md

```diff
@@ -0,0 +1,60 @@
+# Upgrading and Migrating contracts
+
+We consider two approaches:  *Upgrading* existing contracts to a new version but retaining
+state data as-is, and; *migrating* users to a completely new set of contracts.
+
+We would only migrate in the event of a key loss, which prevents upgrading contracts.
+
+## Upgrade contracts approach
+
+Our contracts are already mostly divided into contracts that contain data and contracts that
+contain logic. This is fortuitous for the pattern we are going to use.
+
+`State.sol` contains most of the state, with some in `Shield.sol`.
+This division could be made complete by moving the `proposeBlock` function out of `State.sol`
+and moving the `Shield.sol` data into `State.sol` (there may be a few other minor adjustments
```

Comment:
Minor adjustments would mean removing any function that has logic other than getting or setting a value, such as updateBondAccountTime, removeProposer, rewardChallenger, withdraw, emitRollback?

Reply:
Yes, although `updateBondAccountTime` should probably be considered a setter, and we need to leave `withdraw` there, because `State.sol` holds all of nightfall's money (that also being a type of state), so it's the only place that we can pay out from.